kustomize/commands/edit/set/configmap.go

```diff
@@ -0,0 +1,151 @@
+`,
+		RunE: func(_ *cobra.Command, args []string) error {
+			return runEditSetConfigMap(flags, fSys, args, ldr, rf)
+		},
+	}
+
+	cmd.Flags().StringArrayVar(
+		&flags.LiteralSources,
+		util.FromLiteralFlag,
+		[]string{},
+		"Specify an existing key and a new value to update a ConfigMap (i.e. mykey=newvalue)")
+	cmd.Flags().StringVar(
+		&flags.Namespace,
+		util.NamespaceFlag,
+		"",
+		"Current namespace of the target ConfigMap")
+	cmd.Flags().StringVar(
+		&flags.NewNamespace,
+		util.NewNamespaceFlag,
+		"",
+		"New namespace value for the target ConfigMap")
+
+	return cmd
+}
+
+func runEditSetConfigMap(
+	flags util.ConfigMapSecretFlagsAndArgs,
+	fSys filesys.FileSystem,
+	args []string,
+	ldr ifc.KvLoader,
+	rf *resource.Factory,
+) error {
+	err := flags.ExpandFileSource(fSys)
+	if err != nil {
+		return fmt.Errorf("failed to expand file source: %w", err)
+	}
+
+	err = flags.ValidateSet(args)
+	if err != nil {
+		return fmt.Errorf("failed to validate flags: %w", err)
+	}
+
+	// Load the kustomization file.
+	mf, err := kustfile.NewKustomizationFile(fSys)
+	if err != nil {
+		return fmt.Errorf("failed to load kustomization file: %w", err)
+	}
+
+	kustomization, err := mf.Read()
+	if err != nil {
+		return fmt.Errorf("failed to read kustomization file: %w", err)
+	}
+
+	// Updates the existing ConfigMap
+	err = setConfigMap(ldr, kustomization, flags, rf)
+	if err != nil {
+		return fmt.Errorf("failed to create configmap: %w", err)
+	}
+
+	// Write out the kustomization file with added configmap.
+	err = mf.Write(kustomization)
+	if err != nil {
+		return fmt.Errorf("failed to write kustomization file: %w", err)
+	}
+
+	return nil
+}
+
+func setConfigMap(
+	ldr ifc.KvLoader,
+	k *types.Kustomization,
+	flags util.ConfigMapSecretFlagsAndArgs,
+	rf *resource.Factory,
+) error {
+	args, err := findConfigMapArgs(k, flags.Name, flags.Namespace)
+	if err != nil {
+		return fmt.Errorf("could not set new ConfigMap value: %w", err)
+	}
+
+	if len(flags.LiteralSources) > 0 {
+		err := util.UpdateLiteralSources(&args.GeneratorArgs, flags)
+		if err != nil {
+			return fmt.Errorf("failed to update literal sources: %w", err)
+		}
+	}
+
+	// update namespace to new one
+	if flags.NewNamespace != "" {
+		args.Namespace = flags.NewNamespace
+	}
+
+	// Validate by trying to create corev1.configmap.
+	args.Options = types.MergeGlobalOptionsIntoLocal(
+		args.Options, k.GeneratorOptions)
+
+	_, err = rf.MakeConfigMap(ldr, args)
+	if err != nil {
+		return fmt.Errorf("failed to validate ConfigMap structure: %w", err)
+	}
+
+	return nil
+}
+
+// findConfigMapArgs finds the generator arguments corresponding to the specified
+// ConfigMap name. ConfigMap must exist for this command to be successful.
+func findConfigMapArgs(m *types.Kustomization, name, namespace string) (*types.ConfigMapArgs, error) {
+	cmIndex := slices.IndexFunc(m.ConfigMapGenerator, func(cmArgs types.ConfigMapArgs) bool {
+		return name == cmArgs.Name && namespace == cmArgs.Namespace
```

Comment:
nit: This is an edge case, but I think we should try to handle the default namespace being specified as either as an empty string or default. 

`""` and `"default"` should be treated as the same namespace, so, for example, it should still match if there is no namespace specified in `cmArgs.Namespace` but `flags.Namespace` is `default` and vice versa.

We should also have a test or two for this.

Reply:
That's fair. Will update.

Reply:
I just found out that this is already an issue for `edit add configmap`/`edit add secret`. Will open a separate PR to fix the behaviour on these two other places.

Reply:
I opened #5430 to address this in other places and will hold this one until the other gets merged, so I can reuse the same method.

/hold

Reply:
#5430 was merged, so I applied the changes to this code to handle the empty and the default namespaces as being the same.